spec/flags_spec.rb

```diff
@@ -59,6 +62,12 @@ def capture_stdout
     it { is_expected.to match(/a-file.+z-file.+symlinks/) } # sorts results alphabetically, files first
   end
 
+  context 'with --sort=size flag' do
+    let(:args) { ['--sort=size', FIXTURES] }
+
+    it { is_expected.to match(/a-file.+z-file.+symlinks/) } # sorts results by size
+  end
+
```

Comment:
Add test for `--sort=time` flag too please.

Reply:
I don't have a good idea how this could be done... I'll look into it.

Reply:
Edit the z-file, and commit it. Then, testing for `-t` should be easy right?

Reply:
I think this will not work. The timestamps are not stable for the files, it depends on how git checks them out and what other programs do to them. I think comparing the output from `ls` would be an option.

Reply:
Hmm, let's go with a pending test case for sorting by time for now maybe?

Reply:
This is what I have right now:

```
  context 'with --sort=time' do
    ls_out = `ls --sort=time #{FIXTURES}`.split.each { |line| Regexp.quote(line) }

    expected = Regexp.new ls_out.join('.+')

    let(:args) { ['--sort=time', FIXTURES] }

    it { is_expected.to match(expected) }
  end
```

That's not too bad, IMO. It ensures that colorls basically has the same output for the same flags. This could be extended and somehow refactored into a general test harness / function.

What do you think?

Reply:
Hmm, this seems fine.

Reply:
Something like [this](https://github.com/SciRuby/daru-io/blob/master/spec/daru/io/importers/plaintext_spec.rb#L16) could help for pending test case.

Reply:
If it's an issue with just ruby 2.1, you can add a check for `RUBY_VERSION` and skip only for 2.1.

Reply:
According to this answer on stackoverflow, it seems that indeed Ruby's sort is stable since version 2.2: https://stackoverflow.com/a/44486562

But one should not depend on it...

Reply:
@athityakumar I took a different aproach now. Listing the entries of the fixtures directory, setting a fixed modification time to each entry with one second difference. This seems to work OK...